### PR TITLE
fix: detect targetOrigin if not provided

### PR DESCRIPTION
**Pull request by MayaGillilan**

### CODE REVIEW

File: packages/live-preview-sdk/src/index.ts [hunk 1]
```diff
@@ -133,7 +133,21 @@ export class ContentfulLivePreview {
       this.space = space;
       this.environment = environment;
 
-      this.targetOrigin = Array.isArray(targetOrigin) ? targetOrigin : [targetOrigin];
+      if (targetOrigin) {
+        this.targetOrigin = Array.isArray(targetOrigin) ? targetOrigin : [targetOrigin];
+      } else {
+        const ancestorOrigins = window.location.ancestorOrigins;
+        const currentDefaultOrigin = ancestorOrigins
+          ? DEFAULT_ORIGINS.find((origin) => ancestorOrigins.contains(origin))
```
Comment (chrishelgert) [inline packages/live-preview-sdk/src/index.ts:141]: this are no longedr the default origins and instead the allowed ones?

Comment (chrishelgert) [reply packages/live-preview-sdk/src/index.ts:141]: would it make sense to extract this into a function?

Comment (MayaGillilan) [reply packages/live-preview-sdk/src/index.ts:141]: Hmm true, but they're still sorta defaults in that you can still override them by passing in a targetOrigin to the config. What about `ALLOWED_DEFAULT_ORIGINS`?

Comment (MayaGillilan) [reply packages/live-preview-sdk/src/index.ts:141]: Yeah it's getting a bit long for a constructor so maybe a good idea

**Review by chrishelgert — APPROVED** [review]

**Pull request merged**